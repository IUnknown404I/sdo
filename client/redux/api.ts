import { createApi } from '@reduxjs/toolkit/query/react';
import { getAuthEndpoints } from './endpoints/authEnd';
import { getChatEndpoints } from './endpoints/chatEnd';
import { getUserEndpoints } from './endpoints/userEnd';
import { baseQueryWithRetry } from './utils/apiUtils';
import { getCoursesEndpoints } from './endpoints/coursesEnd';

export const TAG_TYPES_RTK = [
	// auth
	...['Auth', 'Refresh'],
	// users
	...['User', 'User-Meta'],
	// files
	...['Avatars'],
	// messanger
	...['Chats', 'Friends', 'UsersData'],
	// courses
	...['Courses', 'CourseData'],
] as const;
export type TagTypesRTK = (typeof TAG_TYPES_RTK)[number];

export const rtkApi = createApi({
	reducerPath: 'OnyxApi',
	baseQuery: baseQueryWithRetry({ baseUrl: process.env.NEXT_PUBLIC_SERVER, maxRetries: 1 }),
	tagTypes: TAG_TYPES_RTK,
	endpoints: build => {
		return {
			...getAuthEndpoints(build),
			...getUserEndpoints(build),
			...getChatEndpoints(build),
			...getCoursesEndpoints(build),
		};
	},
});

export interface AccessTokenI {
	access_token: string;
	expires_in: string;
}

export interface LoginFullObjI extends AccessTokenI {
	lastPages?: string[];
}
