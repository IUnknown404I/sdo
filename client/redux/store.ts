import { combineReducers, configureStore, ThunkMiddleware } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { useDispatch } from 'react-redux';
import { reducer as formReducer } from 'redux-form';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import thunk from 'redux-thunk';
import { checkProductionMode } from '../utils/utilityFunctions';
import { rtkApi } from './api';
import rootSaga from './sagas/rootSaga';
import { accessTokenSlice } from './slices/accessToken';
import { authSlice } from './slices/auth';
import { axiosInstanceSlice } from './slices/axiosInstance';
import { messengerSlice } from './slices/messenger';
import { userSlice } from './slices/user';
import { coursesSlice } from './slices/courses';

export const rootEndpoint = [rtkApi];

const rootReducer = combineReducers({
	form: formReducer,
	// redux
	auth: authSlice.reducer,
	user: userSlice.reducer,
	accessToken: accessTokenSlice.reducer,
	axiosInstance: axiosInstanceSlice?.reducer,
	messenger: messengerSlice.reducer,
	courses: coursesSlice.reducer,
	// api
	[rtkApi.reducerPath]: rtkApi.reducer,
	// ...rootEndpoint.reduce((accum, endpoint) => {
	// 	accum[endpoint.reducerPath as string] = endpoint.reducer;
	// 	return accum;
	// }, {} as { [key: string]: typeof rtkApi.reducer }),
});

export type ExtraArgumentsOptions = { productionMode: boolean };
const extraArgument: ExtraArgumentsOptions = {
	productionMode: checkProductionMode(),
};

const sagaMiddleware = createSagaMiddleware();
export const store = configureStore({
	reducer: rootReducer,
	middleware: getDefaultMiddleware =>
		getDefaultMiddleware({
			thunk: { extraArgument },
			serializableCheck: {
				// Ignore these action types
				// ignoredActions: ['auth/fetchAndUpdateTokens'],
				// Ignore these field paths in all actions
				// ignoredActionPaths: ['auth/fetchAndUpdateTokens'],
				// Ignore these paths in the state
				ignoredPaths: ['axiosInstance.instance'],
			},
		})
			.prepend()
			.concat(thunk)
			.concat(sagaMiddleware)
			.concat(rtkApi.middleware)
			// .concat(rootEndpoint.map(endpoint => endpoint.middleware as ThunkMiddleware))
			.concat(extraArgument.productionMode ? [] : [logger as ThunkMiddleware]),
});
// optional, but required for refetchOnFocus/refetchOnReconnect behaviors
setupListeners(store.dispatch);
sagaMiddleware.run(rootSaga);

// Infer the `RootState` and `AppDispatch` types from the root Reducer itself
export type RootState = ReturnType<typeof rootReducer>;
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootStoreState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
