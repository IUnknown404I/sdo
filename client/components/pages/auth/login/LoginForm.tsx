import { useTheme } from '@mui/material/styles';
import { useFormik } from 'formik';
import { useRouter } from 'next/router';
import React, { MouseEventHandler } from 'react';
import * as yup from 'yup';
import useLoading from '../../../../hooks/useLoading';
import { LoginFullObjI } from '../../../../redux/api';
import { useAppDispatch } from '../../../../redux/store';
import AuthThunks from '../../../../redux/thunks/auth';
import { OnyxTypography } from '../../../basics/OnyxTypography';
import { errorMessage, notification } from '../../../utils/notifications/Notification';
import AuthModal from '../components/AuthModal';
import LoginElement from './LoginElement';

export interface AuthLogindataI {
	username: string;
	password: string;
	email: string;
}

const LoginForm = () => {
	const theme = useTheme();
	const router = useRouter();
	const dispatcher = useAppDispatch();

	// where 1 - 4 are attempts; 0 - -infinity is a blocked state and 999 is an inactive account.
	const [attempts, setAttepmts] = React.useState<number>(5);
	const [emailMode, setEmailMode] = React.useState(false);
	const [dataSave, setDataSave] = React.useState<boolean>(true);
	const [showPassword, setShowPassword] = React.useState(false);
	const { Loader, state: isLoading, setState: setIsLoading } = useLoading({ iconVariant: true });
	const [sourceModalState, setSourceModalState] = React.useState<boolean>(false);
	const [blockModalState, setBlockModalState] = React.useState<{ state: boolean; blockReason?: string }>({
		state: false,
	});

	const cachedUsernameOrEmail = React.useRef({ username: '', email: '' });

	const formik = useFormik({
		initialValues: {
			username: '',
			password: '',
			email: 'Disable@dis.no',
		},
		validationSchema: getValidationScheme(emailMode),
		onSubmit: async values => {
			setIsLoading(true);
			dispatcher(AuthThunks.authorize({ ...values, dataSave }))
				.then(res =>
					validateLoginResponce(res.payload as string | LoginFullObjI, values.username, values.email),
				)
				.catch(err => {
					notification({
						message: errorMessage(err) || 'Не удалось авторизоваться! Проверьте данные и попробуйте снова.',
						type: 'error',
						autoClose: 4000,
						theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
					});
				})
				.finally(() => setIsLoading(false));
		},
	});

	React.useEffect(() => {
		if (!router.isReady) return;
		if (router.query.source && !Array.isArray(router.query.source)) setSourceModalState(true);
	}, [router.isReady]);

	return (
		<>
			<LoginElement
				{...{
					dataSave,
					emailMode,
					Loader,
					loadingState: isLoading,
					setDataSave: (state: boolean) => setDataSave(() => state),
					showPassword,
					formik,
					attempts,
					handleAdditionalRecoveryRequest,
					handleClickShowPassword: () => setShowPassword(show => !show),
					handleClickEmailMode: handleClickEmailMode as MouseEventHandler<HTMLButtonElement>,
					handleMouseDownPassword: (event: React.MouseEvent<HTMLButtonElement>) => event.preventDefault(),
				}}
			/>

			<AuthModal
				uncontrolled
				state={sourceModalState}
				setState={() => setSourceModalState(prev => !prev)}
				title={
					router.query.source === 'validation'
						? 'Учетная запись активирована!'
						: 'Данные учетной записи изменены!'
				}
				disableButton
			>
				<p>
					{router.query.source === 'validation'
						? 'Ваша учетная запись успешно активирована! Теперь вам предоставлен доступ к образовательной платформе. Войдите в свой личный кабинет, заполнив форму авторизации.'
						: 'Данные учетной записи были успешно изменены! В целях безопасности вы были отключены от системы. Пожалуйста, войдите в свой личный кабинет, чтобы увидеть изменения.'}
				</p>
			</AuthModal>
			
			<AuthModal
				uncontrolled
				state={blockModalState.state}
				setState={() => setBlockModalState({ state: false, blockReason: undefined })}
				title='Учётная запись заблокирована!'
				disableButton
			>
				<p>
					Ваша учётная запись пользователя была заблокирована!
					<br />
					Обратитесь в техническую поддержку.
					<br />
					{blockModalState?.blockReason && (
						<OnyxTypography
							component='span'
							tpSize='inherit'
							tpWeight='bold'
							sx={{ textDecoration: 'unset' }}
							text={blockModalState.blockReason}
						/>
					)}
				</p>
			</AuthModal>
		</>
	);

	/**
	 * @IUnknown404I
	 * The function for displaing toast according passed response and control the error messages for Login form.
	 * @param payload as [object]: AccessTokenI & { lastPages?: string } or you can pass an error string for displaying corresponding info.
	 */
	function validateLoginResponce(payload: string | LoginFullObjI, username?: string, email?: string): void {
		// string equals to error message => handling errors
		if (typeof payload === 'string') {
			if (payload.trim().includes('Attempts left: ')) {
				const avaibleAtemptsCount = payload.trim().split('Attempts left: ')[1];
				setAttepmts(() => +avaibleAtemptsCount);
				notification({
					message: `Указаны неправильные данные! Попробуйте снова, еще доступно ${avaibleAtemptsCount} ${
						avaibleAtemptsCount === '1' ? 'попытка' : 'попытки'
					}.`,
					type: 'warning',
					autoClose: 6500,
					theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				});
				return;
			} else if (
				payload.toLowerCase().trim().includes('заблокирован') ||
				payload.toLowerCase().trim().includes('blocked')
			) {
				setAttepmts(() => 0);
				notification({
					message: payload,
					type: 'error',
					autoClose: 4000,
					theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				});
				setBlockModalState({ state: true, blockReason: payload.trim().split('!')[1].trim() });
			} else if (
				payload.toLowerCase().trim().includes('не активирована') ||
				payload.toLowerCase().trim().includes('not activ') ||
				payload.toLowerCase().trim().includes('inactive')
			) {
				if ((username && username !== 'DisabledValue*0') || (email && email !== 'Disable@dis.no'))
					cachedUsernameOrEmail.current = {
						username: username || 'DisabledValue*0',
						email: email || 'Disable@dis.no',
					};
				setAttepmts(() => 9999);
				notification({
					message: payload,
					type: 'error',
					autoClose: 4000,
					theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				});
			} else {
				setAttepmts(() => 5);
				notification({
					message: 'Не удалось авторизоваться! Проверьте данные и попробуйте снова.',
					type: 'error',
					autoClose: 4000,
					theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				});
			}
		} else {
			notification({
				message: 'Вы успешно авторизовались в системе дистанционного образования!',
				type: 'success',
				autoClose: 6000,
				theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
			});
			setAttepmts(() => 5);
			formik.resetForm();
			formik.setValues({ username: '', email: '', password: '' });
		}
	}

	function handleAdditionalRecoveryRequest(): void {
		if (!cachedUsernameOrEmail.current.email && !cachedUsernameOrEmail.current.username) return;
		dispatcher(AuthThunks.additionalRecoveryRequest(cachedUsernameOrEmail.current)).then(res =>
			typeof res.payload === 'boolean' && res.payload
				? notification({
						message: 'Заявка успешно отправлена! Проверьте Вашу почту.',
						type: 'success',
						autoClose: 6000,
						theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				  })
				: notification({
						message: res.payload as string,
						type: 'error',
						autoClose: 4000,
						theme: theme.palette.mode === 'dark' ? 'dark' : 'light',
				  }),
		);
	}

	function handleClickEmailMode() {
		if (!emailMode) {
			formik.values.email = formik.values.username;
			formik.values.username = 'DisabledValue*0';
		} else {
			formik.values.username = formik.values.email;
			formik.values.email = 'disable@dis.no';
		}
		setEmailMode(mode => !mode);
	}
};

function getValidationScheme(emailMode?: boolean) {
	return !emailMode
		? yup.object({
				username: yup
					.string()
					.trim()
					.min(5, 'Минимальная длина логина 5 символов!')
					.max(64, 'Достигнута максимальная длина.')
					.required('Необходимо ввести имя пользователя!'),
				password: yup
					.string()
					.min(12, 'Минимальная длина пароля 12 символов!')
					.max(64, 'Достигнута максимальная длина.')
					.required('Необходимо ввести пароль!'),
		  })
		: yup.object({
				email: yup
					.string()
					.email('Некорректно введена электронная почта!')
					.min(6, 'Минимальная длина электронной почты 6 символов!')
					.max(64, 'Достигнута максимальная длина.')
					.required('Необходимо ввести электронную почту!'),
				password: yup
					.string()
					.min(12, 'Минимальная длина пароля 12 символов!')
					.max(64, 'Достигнута максимальная длина.')
					.required('Необходимо ввести пароль!'),
		  });
}

export default LoginForm;
