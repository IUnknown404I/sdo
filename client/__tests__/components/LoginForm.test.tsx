/**
 * @jest-environment jsdom
 */

import { getByText, render, waitFor } from '@testing-library/react';
import React from 'react';
import LoginForm from '../../components/pages/auth/login/LoginForm';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';

describe('LoginForm ->', () => {
	const { container } = render(
		<Provider store={store}>
			<LoginForm />
		</Provider>,
	);

	it('correct renders', () => {
		waitFor(() => expect(container).toMatchSnapshot());
	});

	it('content validity check', () => {
		waitFor(() => expect(getByText(container, 'АВТОРИЗАЦИЯ')).toBeInTheDocument());
		waitFor(() => expect(getByText(container, 'Введите логин')).toBeInTheDocument());
		waitFor(() => expect(getByText(container, 'Введите пароль')).toBeInTheDocument());
		waitFor(() => expect(getByText(container, 'Забыли пароль')).toBeInTheDocument());
		waitFor(() => expect(getByText(container, 'Ещё нет аккаунта?')).toBeInTheDocument());
		waitFor(() => expect(getByText(container, 'Запомнить меня')).toBeInTheDocument());
	});
});
