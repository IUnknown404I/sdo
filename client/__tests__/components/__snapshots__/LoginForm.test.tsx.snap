// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm -> correct renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1xmam07"
  >
    <div
      class="MuiBox-root css-xoaw21"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-1y5fyax-MuiPaper-root"
      >
        <form
          style="display: flex; flex-direction: column; gap: 1rem; border-radius: 20px;"
        >
          <div
            class="MuiBox-root css-1snj1at"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-alignLeft css-uphzpi-MuiTypography-root"
            >
              АВТОРИЗАЦИЯ
            </h2>
            <img
              alt="Authorization"
              height="75"
              src="/icons/auth.svg"
              width="75"
            />
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
            >
              Введите логин
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="username"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle login or email mode"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="Использовать почту"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-mui-internal-clone-element="true"
                    data-testid="AccountCircleRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Введите логин
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
            >
              Введите пароль
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="Показать пароль"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-mui-internal-clone-element="true"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Введите пароль
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-el6gaj"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1net5pt-MuiFormControlLabel-root"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight css-1glyh4b-MuiTypography-root"
              >
                Запомнить меня
              </p>
            </label>
            <span
              aria-label="Восстановить доступ"
              class=""
              data-mui-internal-clone-element="true"
            >
              <a
                href="/recovery"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight css-vssuxv-MuiTypography-root"
                >
                  Забыли пароль?
                </p>
              </a>
            </span>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-p7oec8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Войти
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-pz3r64-MuiCircularProgress-root"
              role="progressbar"
              style="width: 24px; height: 24px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate MuiCircularProgress-circleDisableShrink css-79nvmn-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="18"
                  stroke-width="8"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-fv2zs6"
          >
            <div
              class="MuiBox-root css-9elqfl"
            >
              <div
                aria-label="Показать справку"
                class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-1vb8bex-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                Аккаунт заблокирован?
              </div>
            </div>
            <div
              class="MuiBox-root css-134wu7r"
            >
              <a
                aria-label="Регистрация"
                class=""
                data-mui-internal-clone-element="true"
                href="/registration"
              >
                <div
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-alignRight css-nccj0v-MuiTypography-root"
                >
                  Ещё нет аккаунта?
                </div>
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-1wgvx8i-MuiTypography-root"
    >
      © Система дистанционного образования научно-образовательного центра ООО «Газпром межрегионгаз инжиниринг», 2023
    </h3>
  </div>
</div>
`;
